Automattic/wp-calypso
# Migrate Privacy Policy Banner to use requestHttpData

Stop using data layer to do the simple `/privacy-policy` request and use `requestHttpData` instead. Refactors and simplifies the whole component.

Also lets us remove the `privacyPolicy` reducer and data layer handlers.

Part of #25121.

**How to test:**
1. Start Calypso with `privacy-policy/test` flag enabled. I.e., append `?flags=privacy-policy/test` query param to your Calypso URL.
2. Go to `/stats/day/:site` for some site.
3. Verify that the Privacy Banner is displayed.
4. In Network Devtool, verify that the `/privacy-policy` REST request was sent.
5. Click on "Learn More" and then close the dialog with privacy policy text.
6. In Network Devtool, verify that the `privacy-policy` preference was saved, recording that you have seen the policy.

**Bundle size impact:**
Not bad:
```
chunk     stat_size           parsed_size           gzip_size
build       -3279 B  (-0.1%)      -1403 B  (-0.1%)     -280 B  (-0.1%)
manifest       +0 B                  +0 B                -1 B  (-0.0%)
stats       -2027 B  (-0.4%)       -484 B  (-0.2%)      -49 B  (-0.1%)
```

**One additional bugfix for `Banner` component**
There's a drive-by commit that fixes a bug where a `Banner` that has a "call to action" button is rendered as a `Card` with `is-clickable` CSS class. Then the whole banner area shows a pointer cursor although only the "call to action" button is supposed to be clickable:

<img width="661" alt="privacy-card-cursor" src="https://user-images.githubusercontent.com/664258/43452532-fa4b1b36-94b7-11e8-9323-754482870b48.png">

But only the button should be clickable. This patch fixed that by passing a null `onClick` handler instead of `noop`.
